### ISSUE TITLE
Two False positives SA_LOCAL_SELF_COMPARISON + ES_COMPARING_STRINGS_WITH_EQ  when using instanceof pattern matching (jdk16)

### ISSUE THREAD

**Issue by GandalfTheWhite80** [spotbugs/spotbugs#1617]
When I ran SpotBugs (4.3.1/eclipse 2021-06) with JDK 16.0.1 there are two false positives

SA_LOCAL_SELF_COMPARISON
ES_COMPARING_STRINGS_WITH_EQ 

```java
package de.sph.test;
public class Main {

	public static void main(String[] args) {
		Object o = new Object();
		testInstnaceOfString(o);
	}

	private static boolean testInstnaceOfString(final Object o)
	{
               // SpotBugs SA_LOCAL_SELF_COMPARISON + ES_COMPARING_STRINGS_WITH_EQ  message here
		if (o instanceof final String s) {
			System.out.println("string:" + s);
			return true;
		}		
		return false;
	}
}
```

Is this only a problem of the eclipse plugin?

**Comment by javlock** [comment]
I have a similar situation, only I have 1 warning in that place (of my code): SA_LOCAL_SELF_COMPARISON

SOFTWARE:
com.github.spotbugs:spotbugs-maven-plugin:4.5.0.0
com.github.spotbugs:spotbugs:4.5.2

maven:
Apache Maven 3.8.3 (NON_CANONICAL)
Maven home: /opt/maven
Java version: 17.0.1, vendor: N/A, runtime: /usr/lib/jvm/java-17-openjdk
Default locale: ru_RU, platform encoding: UTF-8
OS name: "linux", version: "5.15.8-arch1-1", arch: "amd64", family: "unix"

java:
openjdk version "17.0.1" 2021-10-19
OpenJDK Runtime Environment (build 17.0.1+12)
OpenJDK 64-Bit Server VM (build 17.0.1+12, mixed mode)

**Comment by javlock** [comment]
@GandalfTheWhite80 this is the error of spotbugs itself

**Comment by PikeTec** [comment]
Just wanting to add:
When using boxed primitives like `if (o instanceof Double d) { ...` you get similar false positive "RC_REF_COMPARISON" and a false positive self comaprison.

**Comment by nbauma109** [comment]
is the issue still there ?

**Comment by GandalfTheWhite80** [comment]
No - I cannot reprocude it anymore with SpotBugs (Eclipse Plugin) 4.8.3
